resources/js/components/news/NewsDetail.vue

```diff
@@ -0,0 +1,134 @@
+        name: "NewsDetail",
+        mounted() {
+            let app = this;
+            let id = window.location.href.split('/').pop();
+            app.newsId = id;
+            axios.get(laroute.route('api.news.show', {news: id}))
+                .then(function (resp) {
+                    app.news = resp.data;
+                })
+                .catch(function () {
+                    alert("Could not load your news")
+                });
+        },
+        async created() {
+            try {
+                let id = window.location.href.split('/').pop();
+                const res = await axios.get(laroute.route('news.comment', {id: id}));
+                const res1 = await axios.get(laroute.route('api.comment.show', {comment: id}));
+                this.cmts = res.data;
+                this.data = res1.data;
+                this.$data.comment.id_news = this.cmts[0].id;
+                this.$data.comment.id_user = this.cmts[1].id;
+            } catch (e) {
+                console.error(e)
+            }
+        },
+        data: function () {
+            return {
+                newsId: null,
+                news: {
+                    title: '',
+                    content: '',
+                    posted_by: '',
+                    created_at: '',
+                },
+                comment: {
+                    id_user: '',
+                    id_news: '',
+                    comment: '',
+                    created_at: ''
+                },
+                cmts: [],
+                data: [],
+                index: laroute.route('news.index'),
+            }
+        },
+        methods: {
+            saveForm() {
+                event.preventDefault();
+                var app = this;
+                var newComment = app.comment;
+                let id = window.location.href.split('/').pop();
+                axios.post(laroute.route('api.comment.store'), newComment)
+                    .then(function (resp) {
+                        window.location = laroute.route('news.detail', {id: id});
+                    })
+                    .catch((error) => {
+                        this.errors = error.response.data.errors;
+                        console.log(response.data.errors);
+                    });
+            },
+            showAlertConfirm(id, index) {
+                this.$swal({
+                    title: 'Are you sure?',
+                    text: "You won't be able to revert this!",
+                    type: 'warning',
+                    showCancelButton: true,
+                    confirmButtonColor: '#3085d6',
+                    cancelButtonColor: '#d33',
+                    confirmButtonText: 'Yes, delete it!'
+                }).then((result) => {
+                    if (result.value) {
+                        var app = this;
+                        axios.delete(laroute.route('api.comment.destroy', {comment: id}))
+                            .then(function (resp) {
```

Comment:
sử dụng arrow function. thì em có thể viết this trong function bình thường nhé không phải đặt `var app = this;` đâu